content/exchange/artifacts/IRIS.Timeline.Add.yaml

```diff
@@ -0,0 +1,219 @@
+                    else= if(condition=Category=~"^rem",then=3,
+                    else= if(condition=Category=~"^ini|^ta0001$",then=4,
+                    else= if(condition=Category=~"^exec|^ta0002$",then=5,
+                    else= if(condition=Category=~"^pers|^ta0003$",then=6,
+                    else= if(condition=Category=~"^priv|^ta0004$",then=7,
+                    else= if(condition=Category=~"^def|^ta0005$",then=8,
+                    else= if(condition=Category=~"^cred|^ta0006$",then=9,
+                    else= if(condition=Category=~"^disc|^ta0007$",then=10,
+                    else= if(condition=Category=~"^lat|^ta0008$",then=11,
+                    else= if(condition=Category=~"^coll|^ta0009$",then=12,
+                    else= if(condition=Category=~"^c2|^com|^ta0011$",then=13,
+                    else= if(condition=Category=~"^exf|^ta0010$",then=14,
+                    else= if(condition=Category=~"^imp|^ta0040$",then=15,
+                    else= 1))))))))))))))
+                    
+      LET eventColor = if(condition=Color =~ "^white",then="#fff",
+                else = if(condition=Color =~ "^blue",then="#1572E899",
+                else = if(condition=Color =~ "^purple",then="#6861CE99",
+                else = if(condition=Color =~ "^lightblue",then="#48ABF799",
+                else = if(condition=Color =~ "^red",then="#F2596199",
+                else = if(condition=Color =~ "^orange",then="#FFAD4699",
+                else = "#31CE3699"))))))
+                
+      LET eventDate = format(format="%vT%v.000",args=[split(string=Timestamp,sep=" ")[0],split(string=Timestamp,sep=" ")[1]])
+      
+      LET eventProperties = serialize(
+                            item=dict(
+                                event_title=Title, 
+                                event_source=if(condition=FlowId,then=format(format="Velo: %v",args=[FlowId]),else="Velo"),
+                                event_assets=if(condition=eventAsset1 OR eventAsset1,then=array(a=eventAsset1,b=eventAsset2),else=[]),
+                                event_iocs=if(condition=IocId,then=split(string=IocId,sep=",|;"),else=[]),
+                                event_tags=if(condition=Tags,then=format(format="Velo,%v",args=[Tags]),else="Velo"),
+                                event_category_id=eventCategory,
+                                event_in_summary=AddToSummary,
+                                event_in_graph=AddToGraph,
+                                event_color=eventColor,
+                                event_date=eventDate,
+                                event_tz="+00:00",
+                                event_content=Description,
+                                event_raw=RawEvent
+                            )
+                            ,format="json"
+                        )
+      
+      LET apiRequestIrisAddEvent = if(condition=config.version.version < "0.6.9",
```

Comment:
it is better to do version camparisons on the plugin itself 

```version(plugin="http_client") < 2```

because comparing on the semantic version is not so simple

Reply:
I just tested this approach, but I get version '2' for both, 0.6.8-2 and 0.6.9. Is this intended? I thought as the parameter skip_verify was added and disable_ssl_security marked as deprecated in 0.6.9, the version of the plugin should be increased as well.
![image](https://github.com/Velocidex/velociraptor-docs/assets/29574367/7fb20d49-09b8-44f8-b000-cfdef26d5f2d)
![image](https://github.com/Velocidex/velociraptor-docs/assets/29574367/80016c91-32f2-4ad3-b07a-04f8e8be8659)

Reply:
The version relates to the version of the plugin itself - in this case 2. When a plugin changes functionality in an incompatible way (e.g. adding new option) then its version is incremented independently of the release version. This makes it easier to write VQL that works with whatever release as long as the plugin needed is of the right version.